drivers/gpio/gpio_nct38xx_port.c

```diff
@@ -146,6 +146,78 @@ static int gpio_nct38xx_pin_config(const struct device *dev, gpio_pin_t pin, gpi
 	return ret;
 }
 
+#ifdef CONFIG_GPIO_GET_CONFIG
+int gpio_nct38xx_pin_get_config(const struct device *dev, gpio_pin_t pin, gpio_flags_t *flags)
+{
+	const struct gpio_nct38xx_port_config *const config = dev->config;
+	struct gpio_nct38xx_port_data *const data = dev->data;
+	uint32_t mask = BIT(pin);
+	uint8_t reg;
+	int ret;
+
+	k_sem_take(&data->lock, K_FOREVER);
+
+	if (config->gpio_port == 0) {
+		if (mask & (~config->common.port_pin_mask)) {
+			ret = -ENOTSUP;
+			goto done;
+		}
+
+		ret = nct38xx_reg_read_byte(config->nct38xx_dev, NCT38XX_REG_MUX_CONTROL, &reg);
+		if (ret < 0) {
+			goto done;
+		}
+
+		if ((mask & config->pinmux_mask) && (mask & (~reg))) {
```

Comment:
Can simplify to:

mask & config->pinmux_mask & ~reg

How about adding a comment too?